src/state/Maps/reducers.js

```diff
@@ -31,6 +31,72 @@ const setActiveMapKey = (state, mapKey) => {
 	}
 };
 
+/**
+ * Add map to store
+ * @param state {Object}
+ * @param map {Object} map state
+ * @return {Object} Updated state
+ */
+const addMap = (state, map) => {
+	if (map?.key) {
+		return {
+			...state,
+			maps: {
+				...state.maps,
+				[map.key]: map,
+			},
+		};
+	} else {
+		return state;
+	}
+};
+
+/**
+ * Add map set to store
+ * @param state {Object}
+ * @param mapSet {Object} map set state
+ * @return {Object} Updated state
+ */
+const addMapSet = (state, mapSet) => {
+	if (mapSet?.key) {
+		return {
+			...state,
+			sets: {
+				...state.sets,
+				[mapSet.key]: mapSet,
+			},
+		};
+	} else {
+		return state;
+	}
+};
+
+/**
+ * Add map to map set
+ * @param state {Object}
+ * @param mapKey {string}
+ * @param mapSetKey {string}
+ * @return {Object} Updated state
+ */
+const addMapToSet = (state, mapKey, mapSetKey) => {
+	if (mapKey && mapSetKey && state.sets[mapSetKey]) {
+		return {
+			...state,
+			sets: {
+				...state.sets,
+				[mapSetKey]: {
+					...state.sets[mapSetKey],
```

Comment:
What if mapSetKey does not exists in state.sets.

Reply:
it is solved in line 82